packages/flutter/lib/src/semantics/semantics_event.dart

```diff
@@ -93,13 +93,18 @@ abstract class SemanticsEvent {
 /// [1]: https://developer.android.com/reference/android/view/View#announceForAccessibility(java.lang.CharSequence)
 ///
 class AnnounceSemanticsEvent extends SemanticsEvent {
-  /// Constructs an event that triggers an announcement by the platform.
+  /// Constructs an event that triggers an announcement by the platform
+  /// on the provided [viewId].
   const AnnounceSemanticsEvent(
     this.message,
     this.textDirection, {
     this.assertiveness = Assertiveness.polite,
+    this.viewId,
   }) : super('announce');
 
+  /// The id of the view that this announcement is on.
+  final int? viewId;
```

Comment:
can this ever be null? I won't worry too much about breaking this class's API since this is an internal class. would still be good to have some doc around how one would get viewId if they really want to use this class